libs/typings/src/ui/store.ts

```diff
@@ -21,12 +21,12 @@ export interface IUserState<T> {
 }
 
 /**
- * Interface defining a user state store
+ * Interface defining authentication state store
  */
-export interface IUserStore<T> {
+export interface IAuthenticationStore<T> {
   login({ provider, checkRegistered }: Login): void;
   logout(): void;
   restoreSession(): void;
```

Comment:
when should `restoreSession` be called? Since the store is a sigleton doesn't make sense to be called internally?

Reply:
it made sense to call the restore session when the singleton instance is created which typically happens during page refresh

Reply:
do you see others cases where it should be called ?

Reply:
Idk. this is why i've asked. But I can see that you are calling it in the constructor. then I would make it private